src/data.cpp

```diff
@@ -1056,6 +1056,10 @@ namespace OpenBabel
     // Check return value from OpenDatafile
     // Suggestion from Zhiguo Liu
     string fn_open = OpenDatafile(ifs, _filename, _envvar);
+    
+    // Check _subdir directory
```

Comment:
Other than that line this PR is solid I'd say. It fixes few things for windows. We just have to work out the "data dir". I second merging it as is and figuring the relatively small bug with Windows "data dir" later on.